modules/internal/CPtr.chpl

```diff
@@ -110,77 +110,109 @@ module CPtr {
   inline proc =(ref a:c_ptr, b:_nilType) { __primitive("=", a, c_nil); }
 
   pragma "no doc"
-  inline proc _cast(type t, x) where t:c_ptr && x:_nilType {
+  inline proc _cast(type t:c_ptr, x:_nilType) {
```

Comment:
These are great changes.

Makes me wonder though whether we should rewrite _cast calls into PRIM_CAST like we now do for isSubtype, to avoid this long repetitive list.

Reply:
We'd have to do that in resolution, right, because we don't know the types otherwise?

Reply:
That seems like an interesting idea. Parsing : calls as PRIM_CAST would also help with #9967, where the casted-to-type might be generic (so can't be passed to a type fn right now).